specs/_features/peerdas/polynomial-commitments-sampling.md

```diff
@@ -0,0 +1,519 @@
+
+## Cells
+
+### Cell computation
+
+#### `compute_cells_and_proofs`
+
+```python
+def compute_cells_and_proofs(blob: Blob) -> Tuple[
+        Vector[Cell, CELLS_PER_BLOB],
+        Vector[KZGProof, CELLS_PER_BLOB]]:
+    """
+    Compute all the cell proofs for one blob. This is an inefficient O(n^2) algorithm,
+    for performant implementation the FK20 algorithm that runs in O(n log n) should be
+    used instead.
+
+    Public method.
+    """
+    polynomial = blob_to_polynomial(blob)
+    polynomial_coeff = polynomial_eval_to_coeff(polynomial)
+
+    cells = []
+    proofs = []
+
+    for i in range(CELLS_PER_BLOB):
+        coset = coset_for_cell(i)
+        proof, ys = compute_kzg_proof_multi_impl(polynomial_coeff, coset)
+        cells.append(ys)
+        proofs.append(proof)
+
+    return cells, proofs
+```
+
+#### `compute_cells`
+
+```python
+def compute_cells(blob: Blob) -> Vector[Cell, CELLS_PER_BLOB]:
+    """
+    Compute the cell data for a blob (without computing the proofs).
+
+    Public method.
+    """
+    polynomial = blob_to_polynomial(blob)
+    polynomial_coeff = polynomial_eval_to_coeff(polynomial)
+
+    extended_data = fft_field(polynomial_coeff + [0] * FIELD_ELEMENTS_PER_BLOB, ROOTS_OF_UNITY_EXTENDED)
+    extended_data_rbo = bit_reversal_permutation(extended_data)
+    return [extended_data_rbo[i * FIELD_ELEMENTS_PER_CELL:(i + 1) * FIELD_ELEMENTS_PER_CELL]
+            for i in range(CELLS_PER_BLOB)]
+```
+
+### Cell verification
+
+#### `verify_cell_proof`
+
+```python
+def verify_cell_proof(commitment: KZGCommitment,
+                      cell_id: int,
+                      cell: Cell,
+                      proof: KZGProof) -> bool:
+    """
+    Check a cell proof
+
+    Public method.
+    """
+    coset = coset_for_cell(cell_id)
+
+    return verify_kzg_proof_multi_impl(commitment, coset, cell, proof)
+```
+
+#### `verify_cell_proof_batch`
+
+```python
+def verify_cell_proof_batch(row_commitments: Sequence[KZGCommitment],
+                            row_ids: Sequence[int],
+                            column_ids: Sequence[int],
+                            cells: Sequence[Cell],
+                            proofs: Sequence[KZGProof]) -> bool:
+    """
+    Check multiple cell proofs. This function implements the naive algorithm of checking every cell
+    individually; an efficient algorithm can be found here:
+    https://ethresear.ch/t/a-universal-verification-equation-for-data-availability-sampling/13240
+
+    Public method.
+    """
+
+    # Get commitments via row IDs
+    commitments = [row_commitments[row_id] for row_id in row_ids]
+    
+    return all(
+        verify_kzg_proof_multi_impl(commitment, coset_for_cell(column_id), cell, proof)
+        for commitment, column_id, cell, proof in zip(commitments, column_ids, cells, proofs)
+    )
+```
+
+## Reconstruction
+
+### `recover_polynomial`
+
+```python
+def recover_polynomial(cell_ids: Sequence[CellID], cells: Sequence[Cell]) -> Polynomial:
```

Comment:
@dankrad wrote function signature `cells: Sequence[Tuple[CellID, Cell]]` previously and I changed it to `cell_ids: Sequence[CellID], cells: Sequence[Cell]`.

I understand that using `Tuple` makes sense there, but it may make the test format more complicated. Open to discuss.